container.go

```diff
@@ -698,3 +719,33 @@ func (container *container) unregisterCallback() error {
 
 	return nil
 }
+
+// Modifies the System by sending a request to HCS
+func (container *container) Modify(config *ResourceModificationRequestResponse) error {
+	operation := "Modify"
+	title := "HCSShim::Container::" + operation
+
+	if container.handle == 0 {
+		return makeContainerError(container, operation, "", ErrAlreadyClosed)
+	}
+
+	container.handleLock.RLock()
```

Comment:
Missed it on first review, but this lock needs to be help when checking if handle is valid above. Move the lock and deferred unlock to the first line of this function.